actions/treasures.go

```diff
@@ -15,7 +24,51 @@ type treasureReq struct {
 	EthKey          string `json:"EthKey"`
 }
 
+type treasureRes struct {
+	Success string `json:"success"`
+}
+
+func unixMilli(t time.Time) int64 {
+	return t.Round(time.Millisecond).UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond))
+}
+
 // Verifies the treasure and claims such treasure.
 func (t *TreasuresResource) VerifyAndClaim(c buffalo.Context) error {
-	return nil
-}
+	req := treasureReq{}
+	oyster_utils.ParseReqBody(c.Request(), &req)
+
+	host_ip := os.Getenv("HOST_IP")
```

Comment:
I think it would be better to pull the logic from lines 40-44 into a new method in the services/iota_wrappers file.  Lines 40-42 could even be omitted since that iota_wrapper file already has the API instantiated.

Reply:
Should I inicializate iota_wrapper - function ini in iota_wrappers? If yes, how? Or is enaugh only import? Function should I insert into structure IotaService in iota_wrappers - example: VerifyTreasure(Datamap, bool) ?

Reply:
In the iota_wrappers file, you'll add a (lowercase) method called `verifyTreasure`.  

In the iota_wrappers file, in the struct on lines 98-103, you'll add:  
`VerifyTreasure:  verifyTreasure`
https://github.com/oysterprotocol/brokernode/blob/master/services/iota_wrappers.go#L98

Beneath the types declared in lines 42-45 in iota_wrappers, you'll add a type declaration:
`type VerifyTreasure func(types of the parameters you are passing in) (types of the return values)`
https://github.com/oysterprotocol/brokernode/blob/master/services/iota_wrappers.go#L42

In the struct on line 35-40, you'll add the following:
`VerifyTreasure   VerifyTreasure`   
https://github.com/oysterprotocol/brokernode/blob/master/services/iota_wrappers.go#L35

When you want to use it in your file, you'll import "github.com/oysterprotocol/brokernode/services" and then in your variables, you'll add this:  
`var IotaWrapper = services.IotaWrapper` 
Here is an example of this in the job_runner.go file:  
https://github.com/oysterprotocol/brokernode/blob/master/jobs/job_runner.go#L13

Then IotaWrapper will be available in your file and you can do `IotaWrapper.MethodIWantToCall()` etc.

Reply:
@tenisakb @pzhao5 Just so you guys know, all these extra steps just to add a method to the iota_wrappers file is in order for us to be able to mock the method easily.